docs/src/main/asciidoc/smallrye-graphql-client.adoc

```diff
@@ -0,0 +1,353 @@
+to the GraphQL query language, general concepts and server-side development.
+
+
+include::./status-include.adoc[]
+
+The guide will walk you through developing and running a simple application that uses both supported
+types of GraphQL clients to retrieve data from a remote resource, that being a database related to Star Wars.
+It's available at https://graphql.org/swapi-graphql[this webpage] if you want to experiment with it manually.
+The web UI allows you to write and execute GraphQL queries against it.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* less than 15 minutes
+* an IDE
+* JDK 11+ installed with `JAVA_HOME` configured appropriately
+* Apache Maven {maven-version}
+
+== GraphQL client types introduction
+
+Two types of GraphQL clients are supported.
+
+The *typesafe* client works very much like the MicroProfile REST Client adjusted for calling GraphQL endpoints.
+A client instance is basically a proxy that you can call like a regular Java object, but under the hood,
+the call will be translated to a GraphQL operation. It works with domain classes directly.
+Any input and output objects for the operation will be translated to/from their representations
+in the GraphQL query language.
+
+The *dynamic* client, on the other hand, works rather like an equivalent of the JAX-RS client
+from the `javax.ws.rs.client` package. It does not require the domain classes to work, it works with
+abstract representations of GraphQL documents instead. Documents are built using a domain-specific language (DSL).
+The exchanged objects are treated as an abstract `JsonObject`, but, when necessary,
+it is possible to convert them to concrete model objects (if suitable model classes are available).
+
+The typesafe client can be viewed as a rather high-level and more declarative approach designed for ease of use,
+whereas the dynamic client is lower-level, more imperative, somewhat more verbose to use, but allows finer grained
+control over operations and responses.
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+Clone the Git repository: `git clone {quickstarts-clone-url}`, or download an {quickstarts-archive-url}[archive].
+
+The solution is located in the `microprofile-graphql-client-quickstart` {quickstarts-tree-url}/microprofile-graphql-client-quickstart[directory].
+
+== Creating the Maven Project
+
+First, we need a new project. Create a new project with the following command:
+
+[source,bash,subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=microprofile-graphql-client-quickstart \
+    -DclassName="org.acme.microprofile.graphql.client.StarWarsResource" \
+    -Dextensions="resteasy-reactive-jsonb,graphql-client"
```

Comment:
That's something I'm a bit concerned about. I suppose the GraphQL client also uses JSON-B under the hood?

We made Jackson our default Jackson implementation so I'm wondering how we should handle this sort of things.

Reply:
Yes, the client (the server side extension too) uses JSON-B.
And also the public API (the abstraction for a GraphQL query response) exposes `javax.json`: https://github.com/smallrye/smallrye-graphql/blob/1.2.1/client/api/src/main/java/io/smallrye/graphql/client/Response.java#L9
So it might be tricky to get rid of it... :(

Reply:
Yes, on a spec level (MP) it's better to be dependent on other things that is available in the umbrella spec (so JSON-P and JSON-B).